### ISSUE TITLE
Null handling bitmap performance

### ISSUE THREAD

**Issue by gianm** [apache/druid#5569]
I noticed that the null bitmap checks from #5278 are usually implemented like:

```java
        public boolean isNull()
        {
          return nullValueBitmap.get(offset.getOffset());
        }
```

From what I recall, `get` on compressed bitmaps can be quite slow compared to walking an iterator.

**Comment by gianm** [comment]
I believe this is still relevant.

**Comment by b-slim** [comment]
@gianm any idea why a get with an offset is more expensive than walking iterator ?

**Comment by gianm** [comment]
@b-slim I'm not familiar with their implementation enough to know, but I could speculate. Perhaps the compressed bitmap iterators cache some information that needs to be re-computed if you are calling `get` repeatedly.

**Comment by gianm** [comment]
Fixed by #8822.
